### PR TITLE
 Add 'Add-ons' visibility flag in WebExtensionBrowserMenuBuilder

### PR DESCRIPTION
**Pull request by ionutbedregeanu**
For #10559
 Adds `showAddonsInMenu` in WebExtensionBrowserMenuBuilder to allow the option of removing `Add-ons`
item even if another extensions are displayed
Add unit tests to test if 'Add-ons' item visibility is updated properly

### Context
Implement a [browser menu](https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/MVP?node-id=902%3A3557) with "Report site issue" extension but without "Add-ons"

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt [hunk 1]
```diff
@@ -37,34 +38,53 @@ class WebExtensionBrowserMenuBuilder(
     private val store: BrowserStore,
     private val style: Style = Style(),
     private val onAddonsManagerTapped: () -> Unit = {},
-    private val appendExtensionSubMenuAtStart: Boolean = false
+    private val appendExtensionSubMenuAtStart: Boolean = false,
+    private val showAddonsInMenu: Boolean = true
 ) : BrowserMenuBuilder(items, extras, endOfMenuAlwaysVisible) {
 
+    private val finalList = items.toMutableList()
```
Comment (Amejia481) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:45]: I'm bit concern about making the `finalList`global in `WebExtensionBrowserMenuBuilder`, as multiple calls to `build` will persist previous values, for example, on line 78 [finalList[index] = menuItem](https://github.com/mozilla-mobile/android-components/pull/10564/files#diff-dccc8321d3ba7a093a75317f56364660611080c9f3e4e155688ce1bdc7609c96R78), we are altering `finalList` and keeping the altered list between calls to `build`, before we used to creating a new list between calls to `build`, now are keeping previous values between subsequent calls to `build`.

Comment (Amejia481) [reply components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:45]: If the intention is just to share the list between functions, we could pass the list as parameter, where we want to use it.

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt [hunk 2]
```diff
@@ -37,34 +38,53 @@ class WebExtensionBrowserMenuBuilder(
...
+        if (index != -1) {
+            menuItem.setIconTint(
+                (finalList[index] as? WebExtensionPlaceholderMenuItem)?.iconTintColorResource
+            )
+            finalList[index] = menuItem
+        }
+        return index == -1
+    }
+
+    private fun showAddonsMenuItems(
```
Comment (Amejia481) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:83]: nit: maybe `createAddonsMenuItems`, as we are creating a new list, what do you think? :)

**Review by Amejia481 — APPROVED** [review]
From my point of view, this looks good to me!
I'm not sure if @mcarare has any feedback?

**Review by mcarare — APPROVED** [review]
LGTM too!

File: components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt [hunk 3]
```diff
@@ -12,14 +12,14 @@ import mozilla.components.browser.menu.item.BrowserMenuDivider
...
+import mozilla.components.browser.menu.item.WebExtensionBrowserMenuItem
 import mozilla.components.browser.menu.item.WebExtensionPlaceholderMenuItem
 import mozilla.components.browser.state.selector.selectedTab
 import mozilla.components.browser.state.store.BrowserStore
 
 /**
  * Browser menu builder with web extension support. It allows [WebExtensionBrowserMenu] to add
- * web extension browser actions in a nested menu item. If there are no web extensions installed,
- * the web extension menu item would return an add-on manager menu item instead.
+ * web extension browser actions in a nested menu item.
```
Comment (mcarare) [inline components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt:22]: I would keep the part of the comment that mentions the add-on manager and just add that its visibility is depending on a boolean.

**Pull request merged**